.travis.yml

```diff
@@ -10,17 +10,24 @@ script:
     else
       VERSION=$(git log --format=%H -1)
     fi
-  - sbt ++$TRAVIS_SCALA_VERSION 'set version in ThisBuild := "'$VERSION'"' test
+  - sbt ++${TRAVIS_SCALA_VERSION}! 'set version in ThisBuild := "'$VERSION'"' clean test
```

Comment:
Lazy question, what does `!` do?. Also curious, do you need the clean?

Reply:
It forces that Scala version. I don't remember exactly what it was but I found `++scalaVersion` wasn't enough for some build cases when the project has `crossScalaVersions` setting and sub-projects.

https://www.scala-sbt.org/1.x/docs/Cross-Build.html
> Sometimes you might want to force the Scala version switch regardless of the crossScalaVersions values. You can use ++ <version>! with exclamation mark for that.

Reply:
For `clean`, I'm not sure how Travis handles the project after build is done. I added as it's alway safe to have `clean` so that the classes from old build won't have any impact on the new one. Yeah it doesn't matter if the old binary files are not cached.